spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/couchbase/CouchbaseConfiguration.java

```diff
@@ -79,6 +86,10 @@ public ClusterInfo couchbaseClusterInfo() {
 	@Bean
 	@Primary
 	public Bucket couchbaseClient() {
+		if (this.properties.getUsername().isEmpty()
+				|| this.properties.getPassword().isEmpty()) {
+			return couchbaseCluster().openBucket(this.properties.getBucket().getName());
```

Comment:
This condition should be inverted. If RBAC `username` or `password` are empty we should call the openBucket method that takes a bucket password.

Sorry if I missed this earlier.

Reply:
> This condition should be inverted. If RBAC `username` or `password` are empty we should call the openBucket method that takes a bucket password.
> 
> Sorry if I missed this earlier.

Thanks. I inverted it. 👍